pkg/controller/util.go

```diff
@@ -616,3 +625,80 @@ func SetPodPvcAnnotations(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) {
 		}
 	}
 }
+
+// GetPreallocation retuns the preallocation setting for DV, falling back to StorageClass and global setting (in this order)
+func GetPreallocation(client client.Client, dataVolume *cdiv1.DataVolume) bool {
+	// First, the DV's preallocation
+	if dataVolume.Spec.Preallocation != nil {
+		return *dataVolume.Spec.Preallocation
+	}
+
+	cdiconfig := &cdiv1.CDIConfig{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: common.ConfigName}, cdiconfig); err != nil {
+		klog.Errorf("Unable to find CDI configuration, %v\n", err)
+		return defaultPreallocation
+	}
+	if cdiconfig.Status.Preallocation == nil {
+		// No need to bother reading storage class if Preallocation not defined in the config
+		return defaultPreallocation
+	}
+
+	// Second, the config setting for the storage class
+	sc := GetStorageClassNameForDV(client, dataVolume)
+	if sc != "" {
+		if pre, exists := cdiconfig.Status.Preallocation.StorageClass[sc]; exists {
+			return pre
+		}
+	}
+
+	// Third, the global setting
+	if cdiconfig.Status.Preallocation.Global != nil {
+		return *cdiconfig.Status.Preallocation.Global
+	}
+
+	// Finally, the default false
+	return defaultPreallocation
+}
+
+// GetStorageClassNameForDV returns storage class to be used for the DV's PVC
+func GetStorageClassNameForDV(c client.Client, dv *cdiv1.DataVolume) string {
```

Comment:
I see you modified GetStorageClassByName, so you know its there, what is the purpose of this function compared to that one?

Reply:
`GetStorageClassByName` takes a storage class name (string) and returns the storage class (object).
`GetStorageClassNameForDV` takes a DataVolume (object) and returns the storage class name (string) that it uses.

GetStorageClassNameForDV is called to find out what the storage class name is.